### DISCUSSION TITLE
How to get the number of messages (count) from Ready & Unacked

### DISCUSSION THREAD

**Discussion by joe-agent** [rabbitmq/rabbitmq-dotnet-client#1303] [Q&A] [answered]
Hi, I am interested to get number of messages in `Ready` & `Unacked` from the image below:
![image](https://user-images.githubusercontent.com/93580883/220656016-0cba3cf6-ec1d-44e8-86da-69925d6def0c.png)

I could do that by making an API GET call to `localhost:15672/api/vhosts` & get the value from:

- messages
- messages_ready
- messages_unacknowledged
![image](https://user-images.githubusercontent.com/93580883/220656965-4a5fde7c-a9c2-4bb3-b72d-e231a9246932.png)

Is there any built-in method/properties from RabbitMQ.client library to get the values above?
Thanks

**Comment by joe-agent** [comment]
@michaelklishin any advice? Thanks

**Reply by michaelklishin** [reply, marked as answer]
FTR, I find it rude when project maintainers are mentioned directly over a question two minutes after it was asked.

See [`QueueDeclareOk`](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/main/projects/RabbitMQ.Client/client/api/QueueDeclareOk.cs#L37). It returns the number
of messages in Ready state and the number of registered consumers. If you need access more metrics,
use an HTTP API client, or something like Prometheus (and its metrics clients).

**Reply by joe-agent** [reply]
Apologies & thanks for the answer
